### PR TITLE
Deprecate defining password secrets using password hash, function and…

### PR DESCRIPTION
**Pull request by dejanb**
… salt. Additionally fix credentials examples

### CODE REVIEW

File: site/documentation/content/api/management/device-registry-v1.yaml [hunk 1]
```diff
@@ -1036,32 +1043,39 @@ components:
...
-               "pwd-hash": "AQIDBAUGBwg=",
-               "salt": "Mq7wFw==",
-               "hash-function": "sha-512"
-            }]
+            [{
+                auth-id: sensor1,
+                type: hashed-password,
+                secrets: [{
+                             "not-after": "2027-12-24T19:00:00Z",
+                             "pwd-hash": "AQIDBAUGBwg=",
```
Comment (sophokles73) [inline site/documentation/content/api/management/device-registry-v1.yaml]: now that we have deprecated these fields, I do not think that we should use them in example payloads, wouldn't you agree?

File: site/documentation/content/api/management/device-registry-v1.yaml [hunk 2]
```diff
@@ -1036,32 +1043,39 @@ components:
...
-               "pwd-hash": "",
-               "salt": "",
-               "hash-function": ""
+           [{
+               auth-id: sensor1,
+               type: hashed-password,
+               secrets: [{
+                  "id": "349556ea-4902-47c7-beb0-1009ab693fb4",
+                  "not-after": "2027-12-24T19:00:00Z",
+                  "pwd-plain": "",
```
Comment (sophokles73) [inline site/documentation/content/api/management/device-registry-v1.yaml]: see above

File: site/documentation/content/api/management/device-registry-v1.yaml (hunk 1 above)
Comment (dejanb) [reply site/documentation/content/api/management/device-registry-v1.yaml]: I was thinking about that, but decided to leave it in for the time being as we have a lot of examples everywhere with this. I'd revisit all docs and examples as part of the #1565.

Comment (sophokles73) [reply site/documentation/content/api/management/device-registry-v1.yaml]: I see

**Comment by sophokles73** [comment]
@dejanb Oh, before we forget, you should document this in the release notes as well ... under *Deprecations*

**Comment by dejanb** [comment]
> @dejanb Oh, before we forget, you should document this in the release notes as well ... under _Deprecations_

It's already in there. Unless there's some other place I missed?

**Pull request merged**